### PR TITLE
Upgrade to commons 3.0.2 and UFC API (FF-2098)

### PR DESCRIPTION
**Pull request by leoromanovsky**
## motivation

upgrade react-native to the new UFC format and take advantage of upcoming feature releases.

## description

* bump commons to 3.0.2 - this includes the UFC API format changes + drops `axios`. However it does not include changes to the configuration store. We will wait until those are accepted into `js-browser` repo to make those changes here.
* `init` function matches `js-client` except not polling configuration - is this needed?
* removes redundant tests - anything tested in `js-commons` doesn't need to be repeated here
* removes api test server - mocks all remote requests.

## testing

running local ios and android simulators

Screenshot 2024-05-15 at 7 51 10 PM

### CODE REVIEW

File: .github/actions/setup/action.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ runs:
     - name: Setup Node.js
       uses: actions/setup-node@v3
       with:
-        node-version-file: .nvmrc
+        node-version: '18.x'  # Specify the Node.js version directly here
```
Comment (aarsilv) [inline .github/actions/setup/action.yml:10]: :+1:

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@eppo/react-native-sdk",
-  "version": "1.1.0",
+  "version": "3.0.0",
```
Comment (aarsilv) [inline package.json:3]: 📈

File: package.json [hunk 3]
```diff
@@ -51,9 +50,8 @@
     "registry": "https://registry.npmjs.org/"
   },
   "dependencies": {
-    "@eppo/js-client-sdk-common": "2.2.0",
+    "@eppo/js-client-sdk-common": "3.0.2",
```
Comment (aarsilv) [inline package.json:53]: 🔥

File: src/__tests__/index.test.tsx [hunk 4]
```diff
@@ -1,155 +1,124 @@
...
-    });
-    const client_instance = new EppoReactNativeClient(mockConfigStore);
-    client_instance.setLogger(mockLogger);
-    const assignment = client_instance.getAssignment('subject-10', flagKey);
-    expect(assignment).toEqual('variant-2');
+    }) as jest.Mock;
   });
 
-  it('returns null when experiment config is absent', () => {
+  it('returns default value when experiment config is absent', () => {
```
Comment (aarsilv) [inline src/__tests__/index.test.tsx:96]: 👍

File: src/async-storage.spec.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import { EppoAsyncStorage } from './async-storage';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+
+describe('EppoAsyncStorage', () => {
```
Comment (aarsilv) [inline src/async-storage.spec.ts:4]: 🙌

File: src/async-storage.spec.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  const storageKey = '@eppo/sdk-cache-ufc';
```
Comment (aarsilv) [inline src/async-storage.spec.ts:5]: With Android, we encountered users encountering issues changing up API keys on devices. I wonder if we should also make react native storage isolated per API key.

Comment (aarsilv) [reply src/async-storage.spec.ts:5]: Also, should this just be `STORAGE_KEY`?

File: src/async-storage.ts [hunk 7]
```diff
@@ -1,10 +1,12 @@
 import type { IConfigurationStore } from '@eppo/js-client-sdk-common';
 import AsyncStorage from '@react-native-async-storage/async-storage';
 
-const STORAGE_KEY = '@eppo/sdk-cache';
+// Bump this version number when we make breaking changes to the cache.
+const STORAGE_KEY = '@eppo/sdk-cache-ufc';
 
 export class EppoAsyncStorage implements IConfigurationStore {
   private cache: { [key: string]: any } = {};
+  private _isInitialized = false;
```
Comment (aarsilv) [inline src/async-storage.ts:9]: Same as other repo--how come we are using underscore prefix?

**Review by aarsilv — APPROVED** [review]
Good stuff!

File: src/async-storage.ts (hunk 7 above)
Comment (leoromanovsky) [reply src/async-storage.ts:9]: Conflicts with the public method.

**Pull request merged**